contributors/design-proposals/auditing.md

```diff
@@ -0,0 +1,340 @@
+    // Object value after modification (will be empty when removing object)
+    NewObject runtime.Object
+    // Difference between old and new object
+    Patch     []byte
+}
+```
+
+### Policy
+
+The audit policy determines what audit event is generated for a given request. The policy is configured by the cluster administrator. Here is a sketch of the policy API:
+
+```go
+type Policy struct {
+    // Rules specify the audit Level a request should be recorded at.
+    // A request may match multiple rules, in which case the FIRST matching rule is used.
+    // The default audit level is None, but can be overridden by a catch-all rule at the end of the list.
+    Rules []PolicyRule
+
+    // Discussed under Filters section.
+    Filters []Filter
+}
+
+// Based off the RBAC PolicyRule
+type PolicyRule struct {
+    // Required. The Level that requests matching this rule are recorded at.
+    Level Level
+
+    // The users (by authenticated user name) this rule applies to.
+    // An empty list implies every user.
+    Users []string
+    // The user groups this rule applies to. If a user is considered matching
+    // if the are a member of any of these groups
+    // An empty list implies every user group.
+    UserGroups []string
+
+    // The verbs that match this rule.
+    // An empty list implies every verb.
+    Verbs []string
+
+    // Rules can apply to API resources (such as "pods" or "secrets"),
+    // non-resource URL paths (such as "/api"), or neither, but not both.
+    // If neither is specified, the rule is treated as a default for all URLs.
+
+    // APIGroups is the name of the APIGroup that contains the resources ("" for core).
+    // If multiple API groups are specified, any action requested against one of the
+    // enumerated resources in any API group will be allowed.
+    // Any empty list implies every group.
+    APIGroups []string
+    // Namespaces that this rule matches.
+    // This field should be left empty if specifying non-namespaced resources.
+    // Any empty list implies every namespace.
+    Namespaces []string
+    // Resources is a list of resource types this rule applies to.
+    // Any empty list implies every resource type.
+    Resources []string
+    // ResourceNames is an optional white list of names that the rule applies to.
+    // Any empty list implies everything.
+    ResourceNames []string
+
+    // NonResourceURLs is a set of partial urls that should be audited.
+    // *s are allowed, but only as the full, final step in the path.
+    // If an action is not a resource API request, then the URL is split on '/' and
+    // is checked against the NonResourceURLs to look for a match.
+    NonResourceURLs []string
+}
+```
+
+As an example, the administrator may decide that by default requests should be audited with the
+response, except for get and list requests. On top of that, they wish to completely ignore the noisy
+`kube-proxy` endpoint requests. This looks like:
+
+```yaml
```

Comment:
I don't understand the question. Are you asking for an example of the inverse? like
- Default to not logging anything
- Log only secrets requests from non system-account users

(For that last one we'd need to add wildcard matching on usernames)